source/extended-json.rst

```diff
@@ -0,0 +1,618 @@
+  distinction does not exist, such as Javascript.
+
+* ``$numberDouble`` - This is used to preserve the ``double`` type while
+  round-tripping documents to and from extended JSON and to represent special
+  values like NaN or Inf which are not part of the standard JSON specification.
+
+* ``$symbol`` - The use of the ``$symbol`` key preserves the symbol type when
+  round-tripping documents to and from extended JSON.
+
+Timestamp Format
+................
+
+The extended JSON representation for a BSON timestamp changed from its previous
+format of ``{"$timestamp": {"t": <seconds since the epoch>, "i":
+<increment>}}``. The old format did not quote the numeric timestamp and
+increment, making this form difficult to parse on 32-bit systems. Furthermore,
+splitting the timestamp up into two pieces makes the format more difficult to
+generate and parse without any gain, since a BSON timestamp is one number, and
+the increment and timestamp pieces individually are not friendlier for humans to
+read.
+
+Backwards Compatibility
+.......................
+
+Although extended JSON parsers are required to accept all strings in canonical
+extended JSON format, they may accept strings in other formats as well. This
+allows parsers to accept extended JSON emitted by tools that predate this
+specification.
+
+Similarly, generators are permitted to output strings in other formats than
+canonical extended JSON. This allows such a tool to transform a BSON document
+into some interchange format for consumption by a tool that predates this
+specification.
+
+Reference Implementation
+========================
+
+PyMongo implements the canonical extended JSON format, which must be chosen by
+selecting the right option on the ``JSONOptions`` object::
+
+  from bson.json_util import dumps, DatetimeRepresentation, CANONICAL_JSON_OPTIONS
+
+  dumps(document, json_options=CANONICAL_JSON_OPTIONS)
+
+Implementation Notes
+====================
+
+JSON File Format
+----------------
+
+Some applications like mongoexport may wish to write multiple extended JSON
+documents to a single file. One way to do this is to list each JSON document
+one-per-line. When doing this, it is important to ensure that special characters
+like newlines are encoded properly (e.g. ``\n``).
+
+Duplicate Keys
+--------------
+
+The BSON specification allows for duplicate key names within the same BSON
+document. The JSON specification says that names within an object should be
+unique, and many JSON libraries are incapable of handling this scenario. Please
+note there is no defined extended JSON representation for a BSON document that
+contains duplicate key names.
+
+Key Escaping
+------------
+
+There is no escape mechanism for keys that are prefixed with the dollar sign
+(``$``) when parsing extended JSON. In other words, the following code snippets
+produce the same extended JSON string::
+
+  new Document("s", new Document("$symbol": "val"))
+  new Document("s", new Symbol("val"))
```

Comment:
Possibly better phrased along the lines of "the following Java code snippets produce two documents that a generator would to output as the same extended JSON string."  Then show that output as well, and note that when parsed, it would parse to the second form with Symbol.